tests/test_agent_service.py

```
@@ -0,0 +1,216 @@
+    persona = Persona(name="Test Persona", role="Test Role", location="Test Location", background=["bg"],
+                      goals=["goal"], expertise_level={"business": "high", "technology": "low"},
+                      personality={"tone": ["friendly"], "professionalism": "casual",
+                                   "focus": {"can_tangent": False, "refocus_easily": True}},
+                      communication_rules={"avoid": ["jargon"]})
+
+    project = Project(project_name="Test Project", business_summary="summary", requirements=[])
+
+    # patch in mocks
+    monkeypatch.setattr(AgentService, "load_persona", lambda self: setattr(self, "persona", persona))
+    monkeypatch.setattr(AgentService, "load_project", lambda self: setattr(self, "project", project))
+    monkeypatch.setattr(AgentService, "load_history", lambda self, history=None: setattr(self, "history", []))
```

Ruff (0.15.0)

[warning] Unused lambda argument: `history`

(ARG005)